### PR TITLE
Staticcheck: vendor/k8s.io/kubectl/pkg/cmd/set and .../edit

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fix static check failures for:

```
vendor/k8s.io/kubectl/pkg/cmd/set
vendor/k8s.io/kubectl/pkg/cmd/edit
```

Ref #81657

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/set/helper.go [hunk 1]
```diff
@@ -69,7 +70,7 @@ func selectString(s, spec string) bool {
 		// last part does not exactly match remaining part of string
 		case i == (len(parts)-1) && len(s) != (len(part)+next):
 			match = false
-			break
+			break Loop
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/set/helper.go]: Why is this needed, this works just fine and is only needed when two or more loops is in effect.

File: staging/src/k8s.io/kubectl/pkg/cmd/set/set_selector.go [hunk 2]
```diff
@@ -164,6 +164,9 @@ func (o *SetSelectorOptions) RunSelector() error {
 	r := o.ResourceFinder.Do()
 
 	return r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/set/set_selector.go]: This might change behaviour, since we'll break earlier can you double check that?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/kubectl/pkg/cmd/set/helper.go (hunk 1 above)
Comment (gavinfish) [reply staging/src/k8s.io/kubectl/pkg/cmd/set/helper.go]: There is no need to add a break in switch cases for go. I think the logic here is to skip the for loop once the `match` variable is set as false.

File: staging/src/k8s.io/kubectl/pkg/cmd/set/set_selector.go (hunk 2 above)
Comment (gavinfish) [reply staging/src/k8s.io/kubectl/pkg/cmd/set/set_selector.go]: Agree. But I think precheck here makes sense. Similar logic at reconcile https://github.com/kubernetes/kubernetes/blob/5488fe547d0345827d516b81efe4d51762944ddc/pkg/kubectl/cmd/auth/reconcile.go#L192-L197

**Comment by gavinfish** [comment]
PTAL @soltysh

**Review by soltysh — APPROVED** [review]
